### PR TITLE
NAS-131327 / 25.04 / Switch to a local repository

### PR DESCRIPTION
**Pull request by ixhamza**
Switch to a local repository for `smartmontools` to incorporate the fixes for [NAS-131327](https://ixsystems.atlassian.net/browse/NAS-131327) and [NAS-131181](https://ixsystems.atlassian.net/browse/NAS-131181).
Scale Build: http://jenkins.eng.ixsystems.net:8080/job/master/job/custom/701/

### CODE REVIEW

**Review by yocalebo — APPROVED** [review]

File: pull.sh [hunk 1]
```diff
@@ -9,3 +9,10 @@ rm smartmontools_$VERSION-$REVISION.debian.tar.xz
 wget http://deb.debian.org/debian/pool/main/s/smartmontools/smartmontools_$VERSION.orig.tar.xz
 tar xf smartmontools_$VERSION.orig.tar.xz --strip 1
 rm smartmontools_$VERSION.orig.tar.xz
+
+cp NAS-131181.patch debian/patches
+cp NAS-131327.patch debian/patches
+echo 'NAS-131181.patch' >> debian/patches/series
+echo 'NAS-131327.patch' >> debian/patches/series
+
+echo -e "$(cat changelog)\n\n$(cat debian/changelog)" > debian/changelog
```
Comment (amotin) [inline pull.sh]: Are the cat's inside echo a good idea?  If their results are passed to echo command line, then I wonder about command line size limit, escaping, etc.

Comment (ixhamza) [reply pull.sh]: Thanks @amotin. Escaping could indeed be an issue, I have updated it in my latest push.

Comment (amotin) [reply pull.sh]: Thanks. But now I wonder if you need `mv -f` to overwrite the original file?

Comment (ixhamza) [reply pull.sh]: Hmm, it still overwrote without `-f` but I added `-f`.

**Review by amotin — APPROVED** [review]

**Comment by yocalebo** [comment]
backport

**Pull request merged**